graph/graphs/gen/gen_example_test.go

```diff
@@ -0,0 +1,232 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gen_test
+
+import (
+	"fmt"
+	"log"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/encoding/dot"
+	"gonum.org/v1/gonum/graph/graphs/gen"
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+// IDRange is an IDer that provides a set of IDs in [First, Last].
+type IDRange struct{ First, Last int64 }
+
+func (r IDRange) Len() int       { return int(r.Last - r.First + 1) }
+func (r IDRange) ID(i int) int64 { return r.First + int64(i) }
+
+func ExampleStar_undirectedRange() {
+	dst := simple.NewUndirectedGraph()
+	err := gen.Star(dst, 0, IDRange{First: 1, Last: 6})
+	if err != nil {
+		log.Fatal(err)
+	}
+	b, err := dot.Marshal(dst, "star", "", "\t")
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("%s\n", b)
+
+	// Output:
+	// strict graph star {
+	// 	// Node definitions.
+	// 	0;
+	// 	1;
+	// 	2;
+	// 	3;
+	// 	4;
+	// 	5;
+	// 	6;
+
```

Comment:
I could not get these examples to execute until I commented out this newline and others which I'll mark below. After that, they all passed.